jbpm-console-ng-case-mgmt/jbpm-console-ng-case-mgmt-backend/src/test/java/org/jbpm/console/ng/cm/backend/server/RemoteCaseManagementServiceImplTest.java

```diff
@@ -326,4 +332,193 @@ private CaseMilestone createTestMilestone(String caseMilestoneId, String caseMil
 
         return milestone;
     }
+
+    private TaskSummary createTaskSummary(Long taskId, String name, String status) {
+        TaskSummary taskSummary = TaskSummary.builder()
+                .id(taskId)
+                .name(name)
+                .status(status)
+                .build();
+
+        return taskSummary;
+    }
+
+    private CaseAdHocFragment createCaseAdhocFragment(String name, String type) {
+        CaseAdHocFragment caseAdHocFragment = CaseAdHocFragment.builder()
+                .name(name)
+                .type(type)
+                .build();
+
+        return caseAdHocFragment;
+    }
+
+    @Test
+    public void getCaseActionsByStatusTest() {
+        String sortStr="CreatedOn";
+        boolean sortAsc=true;
+        String userId = "userId";
+
+        Long ts1_id = Long.valueOf(1);
+        String ts1_name = "task1";
+        String ts1_status = "Completed";
+        Long ts2_id = Long.valueOf(2);
+        String ts2_name = "task2";
+        String ts2_status = "Suspended";
+
+
+        TaskSummary ts1 = createTaskSummary(ts1_id, ts1_name, ts1_status);
+        TaskSummary ts2 = createTaskSummary(ts2_id, ts2_name, ts2_status);
+
+        when(clientMock.findCaseTasksAssignedAsBusinessAdministrator(eq(caseId), anyString(),
+                anyList(),anyInt(), anyInt(),anyString(),anyBoolean())).thenReturn(Arrays.asList(ts1, ts2));
+
+
+        CaseActionSearchRequest defaultSearchRequest = new CaseActionSearchRequest();
+        defaultSearchRequest.setSort(sortStr);
+        defaultSearchRequest.setSortOrder(sortAsc);
+
+        List<CaseActionSummary> actionSummaries = testedService.getCaseActionsByStatus(caseId,defaultSearchRequest,userId,"Completed","Suspended","Failed",
+                "Error","Exited","Obsolete");
+
+        assertEquals(2,actionSummaries.size());
```

Comment:
I suggest adding one more TaskSummary instance which will not be taken into account in this test (for example with the status "Ready"). Then this assertion would be more valuable.